crates/modelardb_server/src/storage/compressed_data_buffer.rs

```diff
@@ -19,17 +19,46 @@ use std::fs;
 use std::io::Error as IOError;
 use std::io::ErrorKind::Other;
 use std::path::{Path, PathBuf};
+use std::sync::Arc;
 
 use datafusion::arrow::compute;
 use datafusion::arrow::record_batch::RecordBatch;
 use datafusion::parquet::format::SortingColumn;
+use modelardb_common::metadata::model_table_metadata::ModelTableMetadata;
 use modelardb_common::schemas::COMPRESSED_SCHEMA;
 
+use crate::metadata::MetadataManager;
 use crate::storage;
 use crate::storage::StorageEngine;
 
-/// A single compressed buffer, containing one or more compressed segments and providing
-/// functionality for appending segments and saving all segments to a single Apache Parquet file.
+/// Compressed segments representing data points from a column in a model table as one
+/// [`RecordBatch`].
+#[derive(Clone,Debug)]
+pub(super) struct CompressedSegmentBatch {
+    /// Univariate id that uniquely identifies the univariate time series the compressed segments
+    /// represents data points for.
+    pub(super) univariate_id: u64,
+    /// Metadata of the model table to insert the data points into.
+    pub(super) model_table_metadata: Arc<ModelTableMetadata>,
+    /// Metadata of the model table to insert the data points into.
+    pub(super) compressed_segments: RecordBatch,
+}
+
+impl CompressedSegmentBatch {
+    /// Return the name of the table the buffer stores data for.
+    pub(super) fn model_table_name(&self) -> &str {
```

Comment:
It is not necessary but nice to have since it shows the intended use for now, thus I have made the attributes except `compressed_segments` private so they can only be accessed through the methods.